packages/@tinacms/datalayer/src/providers/github.ts

```diff
@@ -0,0 +1,94 @@
+import { Octokit } from '@octokit/rest'
```

Comment:
I thought you were moving all this to @tinacms/graphql?

Reply:
I thought you meant the `createDatabase` function and the gitProvider interface? 

I am OK with moving this but the reason it is here so that folks who are not using self hosted don't need to install the deps of the GithubProvider. If we move it into `@tinacms/graphql` everyone would have to install those extra deps

Reply:
I guess that's a fair point, but I'm not sure how significant that would be and I'm unclear on whether we are going to have a separate self-hosted only dependency or consolidate in either @tinacms/graphql or @tinacms/datalayer. I've been under the impression that we are consolidating, which is the reason why I've been commenting about this stuff getting added to datalayer.

Maybe @octokit/rest should be a peer dependency, since if we have other providers in the future, we may not want to include those either, and they would be peer deps as well.

Reply:
Peers deps does not feel right either since it will still require users to install octokit even if they don't use it. 

Maybe it should be its own package?

Reply:
I think we can mark it as an optional peer dep: https://docs.npmjs.com/cli/v9/configuring-npm/package-json#peerdependenciesmeta

Reply:
We could do that. But if we are going to get the usr to install a bunch of packages would it make sense just to put this in iits own package? That way would could ensure they have the right version of all the dependencies installed

Reply:
Yup makes sense to me! That way we can have one provider per package and we don't have to mix deps.

Reply:
@kldavis4  Just pushed up a change to moves the GithubProvider to its own package called `tinacms-provider-github`